ocaml/xapi/xapi_host_helpers.mli

```diff
@@ -0,0 +1,134 @@
+    false and any pre and post declare_dead scripts will be executed. *)
+
+val consider_enabling_host : __context:Context.t -> unit
+(** [consider_enabling_host ~__context] is called at the end of the xapi
+    startup sequence. It will enable the host unless:
+    {ul
+    {- the user asked the host to be disabled and there was a problem}
+    {- HA is enabled and one-or-more PBDs failed to plug}
+    {- `disabled_until_next_reboot` is set in the local DB}}
+*)
+
+val consider_enabling_host_request : __context:Context.t -> unit
+(** [consider_enabling_host_request ~__context] will ensure that
+    [consider_enabling_host] is called soon. It will coalesce multiple requests
+    that are made. *)
+
+val user_requested_host_disable : bool ref
+(** [user_requested_host_disable] : true if so. Persistent until xapi is
+    restarted *)
+
+val assert_startup_complete : unit -> unit
+(** [assert_startup_complete ()] will raise `host_still_booting` if the startup
+    sequence is not yet complete. *)
+
+module Host_requires_reboot : sig
+  val set : unit -> unit
+  (** [set ()] is used to signal the host needs a reboot. This could be, for
+      example, that the dom0 memory settings have changed and a reboot is
+      required for them to take effect. *)
+
+  val get : unit -> bool
+  (** [get ()] returns [true] if the host needs to be rebooted *)
+end
+
+module Configuration : sig
+  val set_initiator_name : string -> unit
+  (** [set_initiator_name iqn] will write the iscsi initiator configuration to
+      the file specified in Xapi_globs (usually /etc/iscsi/initiatorname.iscsi)
+      *)
+
+  val set_multipathing : bool -> unit
+  (** [set_multipathing enabled] will touch the file specified in Xapi_globs
+      (usually /var/run/nonpersistent/multipath_enabled) if [enabled] is true,
+      otherwise it will remove the file. *)
+
+  val sync_config_files : __context:Context.t -> unit
+  (** [sync_config_files ~__context] ensures that the iscsi iqn and
+      multipathing configuration files reflect the values of the corresponding
+      fields in xapi's database. It should be called at startup on every host
+      BEFORE the other_config watcher [start_watcher_thread] is started *)
+
+  val watch_other_configs : __context:Context.t -> float -> string -> string
+  (** [watch_other_configs ~__context timeout] returns a function that performs
+      one iteration of watching Host.other_config. If an update occurs this
+      will check whether the iscsi_iqn field in other-config is correctly
+      reflected in the field Host.iscsi_iqn, and if not it will call
+      Host.set_iscsi_iqn with the value specified in other-config. This is
+      intended to be run on the master. The returned function has type
+      [token -> token], where [token] is a string. The initial value should be
+      the empty string, and the returned value should be used for further
+      invocations. This function is exposed only for unit testing, and should
```

Comment:
It is annoying that for unit testing we need to do this...

Reply:
There is a ppx to do inline testing, but that would require using yet another ppx...

Reply:
Yes, I know, I wasn't suggesting to do anything. It's unfortunate, I only wish there was some compiler level support for tests behind mli